### PR TITLE
Bumps, enhancements and fixes

### PR DESCRIPTION
**Pull request by f0rmiga**
## Description

- Bumped various external dependencies of the project.
- Enhanced and fixed `yamllint` rules, targets and scripts.

## Motivation and Context

To keep up-to-date with the external dependencies.

## How Has This Been Tested?

By running the bumped external dependencies. Most of them are tested by CI.
The changes to `kind` were tested locally.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code has security implications.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: dev/kind/start.sh [hunk 1]
```diff
@@ -9,15 +9,13 @@ function cluster_exists {
...
+  "${KIND}" create cluster \
+    --name "${CLUSTER_NAME}" \
+    --image "kindest/node:${KUBERNETES_VERSION}"
 else
   echo "Kind is already started"
 fi
 
-# Use the kubeconfig file generated by kind.
-KUBECONFIG="$("${KIND}" get kubeconfig-path --name "${CLUSTER_NAME}")"
-export KUBECONFIG
```
Comment (mook-as) [inline dev/kind/start.sh]: We no longer to use this kubeconfig? Why?

File: dev/kind/def.bzl [hunk 2]
```diff
@@ -41,6 +44,9 @@ attrs = {
     "cluster_name": attr.string(
         mandatory = True,
     ),
+    "kubernetes_version": attr.string(
```
Comment (mook-as) [inline dev/kind/def.bzl]: The minikube branch uses `k8s_version`; please be consistent (or possibly change the minikube one to match).  I think in the minikube case it's also possible to override this with an environment variable that's set when `bazel` is executed… do the same here?

File: dev/kind/start.sh (hunk 1 above)
Comment (f0rmiga) [reply dev/kind/start.sh]: kind is doing like minikube.

Comment (mook-as) [reply dev/kind/start.sh]: I take that to mean "it modifies your default kubeconfig to have a new context &c"? That's fine, then.

Comment (f0rmiga) [reply dev/kind/start.sh]: Yep

**Pull request merged**